### ISSUE TITLE
Support ESM

### ISSUE THREAD

**Issue by aral** [brillout/vite-plugin-ssr#55]
## To reproduce

1. Create the starter Vue app (`npm init vite-plugin-ssr`)
2. Change the project type to ESM (`"type": "module"` in package.json)
3. Update _server.js_ to use import syntax:

   ```js
    import express from 'express'
    import { createPageRender } from 'vite-plugin-ssr/dist'
    import vite from 'vite'
   ```

4. Run it (`npm run dev`)

## What should happen

It should run.

## What actually happens

```
internal/process/esm_loader.js:74
    internalBinding('errors').triggerUncaughtException(
                              ^

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /home/aral/small-tech/small-web/sitekit/sandbox/vite-ssr-project/node_modules/vite-plugin-ssr/index.ts
    at Loader.defaultGetFormat [as _getFormat] (internal/modules/esm/get_format.js:71:15)
    at Loader.getFormat (internal/modules/esm/loader.js:102:42)
    at Loader.getModuleJob (internal/modules/esm/loader.js:231:31)
    at async ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:58:21)
    at async Promise.all (index 1)
    at async link (internal/modules/esm/module_job.js:63:9) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}
```

## Use case

My Node server project uses ESM, not CommonJS.

**Comment by brillout** [comment]
I see what's going on, I'm working on a fix

**Comment by aral** [comment]
Appreciate it – saw you’re almost at v1. Best of luck with the release and thanks for making this; looks great.

**Comment by brillout** [comment]
Fix released in `0.1.0-beta.40`.

> saw you’re almost at v1

It's going to be `0.1.x` for a little while ;-) But yes the plan is to release v1 fairly soon as the interface is stabilizing quickly.

**Comment by aral** [comment]
Awesome, thank you. Looking forward to playing with it more now :)
